mitodl/open-discussions
# Change default parent lookup for nested DRF serializer/url

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #3921

#### What's this PR do?
Adds a new setting, `DRF_NESTED_PARENT_LOOKUP_PREFIX`, that defaults to a blank string, and is used to set the DRF extensions nested parent lookup prefix.

#### How should this be manually tested?
Go to https://localhost:8063/api/v0/schema/swagger-ui/ and find the sections for userlists and stafflists.  You should see something like this with no `{parent_lookup_<staff/userlist>_id}` in sight.

<img src="https://user-images.githubusercontent.com/187676/230381246-9b81ebfa-af47-4116-bfa3-b579ebb59351.png">


Try adding/editing/removing userlists and userlist items, everything should work.  

There's no UI for it yet but you can try the same with stafflists based on these PR instructions.
https://github.com/mitodl/open-discussions/pull/3871
https://github.com/mitodl/open-discussions/pull/3914
